include/revng/Support/OnQuit.h

```diff
@@ -12,17 +12,31 @@
 
 class OnQuitRegistry {
 private:
-  std::vector<std::function<void()>> Registry;
+  struct RegistryEntry {
+    std::function<void()> Handler;
+    bool Repeatable = false;
+  };
+  std::vector<RegistryEntry> Registry;
 
 public:
-  void install();
-
   /// Registers an object for having its onQuit method called upon program
   /// termination
   void add(std::function<void()> &&Handler) {
-    Registry.push_back(std::move(Handler));
+    Registry.push_back({ std::move(Handler), false });
+  }
+
+  /// Registers a function that can be called repeatedly
+  void addRepeatable(std::function<void()> &&Handler) {
+    Registry.push_back({ std::move(Handler), true });
   }
```

Comment:
What problem are you trying to solve specifically?

Reply:
As discussed IRL, I've reworked the API so that users of `OnQuit` can additionally ask to be called on `USR1` and `USR2`